src/main/java/jcmp/DocCompare.java

```diff
@@ -394,6 +393,9 @@ public int hashCode() {
 
         @Override
         public boolean equals(Object x) {
+            if (!(x instanceof DefaultIdentity)) {
```

Comment:
There is cost with creating a stack trace.  It will be faster to not rely on try-catch.